conkit/command_line/conkit_validate.py

```diff
@@ -0,0 +1,253 @@
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+"""This script provides a model validation pipeline. It will
+take a sequence, a predicted distogram and a PDB model, then
+compare the distances observed in the model and those predicted.
+It will then report regions in the model where an outlier was
+detected, and use map_align to provide a solution for potential
+register errors.
+
+It uses one external programs to perform this task:
+
+   map_align for contact map alignment
+
+*** This program needs to be installed separately from https://github.com/sokrypton/map_align***
+
+"""
+
+import argparse
+import os
+import shutil
+from prettytable import PrettyTable
+
+import conkit.applications
+import conkit.command_line
+import conkit.io
+import conkit.plot
+
+logger = None
+
+
+def create_argument_parser():
+    """Create a parser for the command line arguments used in conkit-validate"""
+
+    parser = argparse.ArgumentParser(description=__doc__, formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument("seqfile", type=check_file_exists, help="Path to sequence file")
+    parser.add_argument("seqformat", type=str, help="Sequence format")
+    parser.add_argument("distfile", type=check_file_exists, help="Path to distance prediction file")
+    parser.add_argument("distformat", type=str, help="Format of distance prediction file")
+    parser.add_argument("pdbfile", type=check_file_exists, help="Path to structure file")
+    parser.add_argument("pdbformat", type=str, help="Format of structure file")
```

Comment:
Is it worth setting default values for seqformat, distformat and pdbformat?

Reply:
On a related note, is it worth adding a santity check to make sure that the input files/formats are correct?

Reply:
There's a sanity check downstream at `conkit.io.read` that will raise an "Unrecognised format" exception so I think it's not necessary to do that here.

Reply:
Regarding the default values, I agree that it would make a lot of sense, but since there isn't any default value in the other conkit command line applications I think I will keep it as it is just for consistency.